bspump/main.py

```diff
@@ -1,98 +1,127 @@
 import json
 import os
 import ast
-
-config = None
-__bitswan_dev = False
-__bs_cell_code_contents: dict[int, str] = {}
+import sys
+import tempfile
+import re
 
 from bspump.jupyter import *  # noqa: F403
 import bspump.jupyter
 
-
-def exec_cell(cell, cell_number, ctx):
-    if cell["cell_type"] == "code":
-        source = cell["source"]
-        if len(source) > 0 and "#ignore" not in source[0]:
-            code = (
-                "\n".join(cell["source"])
-                if isinstance(cell["source"], list)
-                else cell["source"]
-            )
-            clean_code = ""
-            for line in code.split("\n"):
-                if line.startswith("!"):
+config = None
+__bitswan_dev = False
+__bs_step_locals = {}
+
+
+def contains_function_call(ast_tree, function_name):
+    for node in ast.walk(ast_tree):
+        if isinstance(node, ast.Call):  # Check if the node is a function call
+            if isinstance(node.func, ast.Name) and node.func.id == function_name:
+                return True
+    return False
+
+
+class NotebookParser:
+    _in_autopipeline = False
+    _cell_number = 0
+    _cell_processor_contents: dict[int, str] = {}
+
+    @classmethod
```

Comment:
Class methods should not be used except for the purpose of creating instances of classes or querying instances of classes and other tasks that cannot be done as normal methods. Here you are setting `_in_autopipeline` as a global variable within the class. This could create thread safety/async safety issues in the future and is surprising and non-standard. Create an instance and use a normal method instead, or don't use a class at all.